api/src/test/resources/io/strimzi/api/kafka/model/JmxTrans-output-definition-with-missing-required-property.yaml

```diff
@@ -9,9 +9,6 @@ spec:
     storage:
       type: persistent-claim
       size: 500Gi
-    listeners:
-      plain: {}
-      tls: {}
```

Comment:
I think these changes might be valid But it would be best to leave them for a separate PR to make the reviews of this one easier and more straight forward.

Reply:
It's because of the testing issues I had, it might be because of the Kubernetes version I'm running the tests against though.

Reply:
I'm fixing the tests by adding patterns to the `anyOf` list of matchers, however, for this one it's a bit tricky 

Here is the error I get if I keep the `listeneres` property
```
`kubectl --namespace kafkacrd-it create -f /Users/f.oueslati/repos/strimzi-kafka-operator/api/target/test-classes/io/strimzi/api/kafka/model/JmxTrans-output-definition-with-missing-required-property.yaml` got status code 1 and stderr:
------
Error from server (BadRequest): error when creating "/Users/f.oueslati/repos/strimzi-kafka-operator/api/target/test-classes/io/strimzi/api/kafka/model/JmxTrans-output-definition-with-missing-required-property.yaml": Kafka in version "v1beta2" cannot be handled as a Kafka: strict decoding error: unknown field "spec.kafka.listeners.plain", unknown field "spec.kafka.listeners.tls" 
```

So, I won't be able to match anything on `outputType` or `name` [as it is expected in the test](https://github.com/strimzi/strimzi-kafka-operator/blob/main/api/src/test/java/io/strimzi/api/kafka/model/KafkaCrdIT.java#L152) 

Should I change the test behaviour to match against `unknown field "spec.kafka.listeners.plain"`? By renaming the test to `testJmxOptionsWithUnknownFields` or should I keep the changes above? Or maybe something else?